### ISSUE TITLE
Tunnels opening

### ISSUE THREAD

**Issue by KingRat0011** [doronz88/pymobiledevice3#940]
MACOS 
Sonoma V 14.3.1

When monitoring this, every time a screenshot is made, it looks like a new tunnel is opened, from the port number that changes?

```
sudo /usr/bin/python3 -m pymobiledevice3 remote tunneld
INFO:     Started server process [4897]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:5555 (Press CTRL+C to quit)
2024-04-15 09:16:12 minimac.local pymobiledevice3.tunneld[4897] INFO Created tunnel --rsd fd2e:71d2:550a::1 58672
2024-04-15 09:16:12 minimac.local pymobiledevice3.tunneld[4897] INFO Created tunnel --rsd fdcc:a6db:a3d::1 52176
INFO:     127.0.0.1:61710 - "GET / HTTP/1.1" 200 OK
INFO:     127.0.0.1:61717 - "GET / HTTP/1.1" 200 OK
INFO:     127.0.0.1:61743 - "GET / HTTP/1.1" 200 OK
INFO:     127.0.0.1:61752 - "GET / HTTP/1.1" 200 OK
INFO:     127.0.0.1:61848 - "GET / HTTP/1.1" 200 OK
INFO:     127.0.0.1:61988 - "GET / HTTP/1.1" 200 OK
INFO:     127.0.0.1:61997 - "GET / HTTP/1.1" 200 OK
```
I take screenshots using the following command
/usr/bin/python3 -m pymobiledevice3 developer dvt screenshot --tunnel $dev_id

But is seems that the MacMini runs out of resources, does it open a new tunnel or port every time a screenshot is taken?

After a while, I cannot open the device screen anymore, but I do not see any errors in the logs, could you please advice?

**Comment by doronz88** [comment]
No, it uses the same one, but it's possible there was an FD leak
Can you verify if this issue is resolved in #938 ?

**Comment by KingRat0011** [comment]
I've upgraded to the latest version,
minimac:~ diginetiot$ pymobiledevice3 version
2.38.1
minimac:~ diginetiot$ pymobiledevice3 version
4.0.0
minimac:~ diginetiot$

Let me test and see if the connectivity is ok after a while

**Comment by KingRat0011** [comment]
It seems more stable with V4.0.0, but I get this in the log
```
/Users/xxxxxxx/Library/Python/3.9/lib/python/site-packages/pymobiledevice3/cli/cli_common.py:250: RuntimeWarning: coroutine 'RemoteServiceDiscoveryService.close' was never awaited
  rsd.close()
RuntimeWarning: Enable tracemalloc to get the object allocation traceback

Sending - /opt/homebrew/bin/python3 -m pymobiledevice3 developer dvt screenshot --tunnel xxx xxx.png
/Users/diginetiot/Library/Python/3.9/lib/python/site-packages/pymobiledevice3/cli/cli_common.py:250: RuntimeWarning: coroutine 'RemoteServiceDiscoveryService.close' was never awaited
  rsd.close()
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

**Comment by doronz88** [comment]
Oh that is a bug I haven't encountered myself. Will create a fix and a release for it in the coming minutes

**Comment by doronz88** [comment]
Just released v4.1.4 which should resolve this issue and another one related to it in the cli.
I'm closing this issue as I understand it is resolved. If any other issues come up please either reopen if it's the same issue, or create another if it's non-related  🙏

**Comment by KingRat0011** [comment]
Thank you!
